result/arrow-parens.js

```diff
@@ -0,0 +1,94 @@
+/*eslint-env es6*/
+
+// Bad
+a => {}
+
+// Good
+(a) => {}
+/*eslint-env es6*/
+
+// Bad
+if (a => 2) {
+}
+
+// Good
+if (a >= 2) {
+}
+/*eslint-env es6*/
+
+// Bad
+(a) => {}
+
+// Good
+a => {}
+/*eslint arrow-parens: ["error", "always"]*/
+/*eslint-env es6*/
+
+a => {};
+a => a;
+a => {'\n'};
+a.then(foo => {});
+a.then(foo => a);
+a(foo => { if (true) {}; });
+/*eslint arrow-parens: ["error", "always"]*/
+/*eslint-env es6*/
+
+() => {};
+(a) => {};
+(a) => a;
+(a) => {'\n'}
+a.then((foo) => {});
+a.then((foo) => { if (true) {}; });
```

Comment:
__[development][eslint]__
* Error - Unexpected parentheses around single function argument (arrow-parens)
* Error - 'foo' is defined but never used (no-unused-vars)
* Error - Unexpected constant condition. (no-constant-condition)
* Error - Empty block statement. (no-empty)
* Error - Unnecessary semicolon. (no-extra-semi)